source/common/router/config_impl.cc

```diff
@@ -120,6 +120,17 @@ RetryPolicyImpl::RetryPolicyImpl(const envoy::config::route::v3::RetryPolicy& re
       }
     }
   }
+
+  if (retry_policy.has_rate_limited_retry_back_off()) {
+    ratelimited_reset_headers_ = Http::HeaderUtility::buildHeaderMatcherVector(
+        retry_policy.rate_limited_retry_back_off().reset_headers());
+
+    ratelimited_reset_max_interval_ =
+        PROTOBUF_GET_OPTIONAL_MS(retry_policy.rate_limited_retry_back_off(), reset_max_interval);
+    if (ratelimited_reset_max_interval_ && ((*ratelimited_reset_max_interval_).count()) < 1) {
+      ratelimited_reset_max_interval_ = std::chrono::milliseconds(1);
+    }
```

Comment:
My thinking here was that in the RetryPolicy this member is optional and doesn't have a default value. So it reflects whatever was configured by the user (possibly nothing). 

Then in RetryStateImpl it has a default defined in the header file. If no value was passed via config nor http header, this default will stand. Otherwise it will be overridden by user input. This test should bear that out:
https://github.com/envoyproxy/envoy/pull/12202/files#diff-0a6a6327b31a217b22c653b1d1ebf4acR914

Reply:
I have a mild preference for doing it here to keep the lookup from the proto and the default together.

Reply:
Yeah I see the value in moving the default to where most of the other defaults live. I've moved it to RetryPolicy now.